### PR TITLE
102 PR2

### PR DESCRIPTION
**Pull request by tapthaker**

### CODE REVIEW

**Comment by jessesquires** [comment]
hey @tapthaker -- sorry, merging your draft at #363 created conflicts. I resolved them though (I think). Be sure to pull this branch down before you make more edits. 😄

File: _drafts/2018-02-08-issue-102.md [hunk 1]
```diff
@@ -73,63 +70,32 @@ metadata, filling in the Int: Hashable requirement as appropriate.
...
-        if (!c2) goto b_loop
-    }
-} else {
-    b_loop:
-    while(c1) {
-        [B]
-    }
-}
-{% endhighlight %}
-
```
Comment (tapthaker) [inline _drafts/2018-02-08-issue-102.md]: @jessesquires : Did you remove this on purpose ? Is so couldn't figure out the reason for it.

**Comment by tapthaker** [comment]
Thanks @jessesquires. Having a problem in indentation after your changes & can't figure out the reason for it. 
screen shot 2018-02-04 at 7 34 14 pm

**Comment by BasThomas** [comment]
I think it will look better now. There were some dangling `>`'s (and I removed them [here](https://github.com/SwiftWeekly/swiftweekly.github.io/pull/364/commits/b9219ae89a88551021af60bf419207d0f5921233)).

Will do a review later today 👍

File: _drafts/2018-02-08-issue-102.md (hunk 1 above)
Comment (jessesquires) [reply _drafts/2018-02-08-issue-102.md]: Ah, yeah. Sorry! I should have mentioned this. I was just trimming down the content 😄

**Comment by jessesquires** [comment]
I think we can really shorten all the Doug Gregor stuff. We really need to keep things short and focused on highlights.

**Comment by tapthaker** [comment]
I agree, my approach was to write everything and to filter via discussing with you folks.
IMO the mangling part is something that might not give insights to the reader and could be removed from the issue.

File: _drafts/2018-02-08-issue-102.md [hunk 2]
```diff
@@ -29,24 +29,74 @@ author: tapthaker
 [Brandon Williams](http://www.fewbutripe.com/about/) and [Stephen Celis](http://www.stephencelis.com/) launched a new Swift video series called [Point-Free](https://www.pointfree.co/). Checkout their first episode, [Functions](https://www.pointfree.co/episodes/ep1-functions).
 >Point-Free is a video series about functional programming and the Swift programming language. Each episode covers a topic that may seem complex and academic at first, but turns out to be quite simple. At the end of each episode we’ll ask “what’s the point?!”, so that we can bring the concepts back down to earth and show how these ideas can improve the quality of your code today.
 
+Brandon's talk [Server-Side Swift from Scratch](https://www.skilled.io/u/swiftsummit/server-side-swift-from-scratch) also discusses some of the functional programming work they have done with Point-Free
```
Comment (BasThomas) [inline _drafts/2018-02-08-issue-102.md]: Nit: let's add a `.` at the end of the sentence.

File: _drafts/2018-02-08-issue-102.md [hunk 3]
```diff
@@ -29,24 +29,74 @@ author: tapthaker
...
+
 [Peter Steinberger](https://twitter.com/steipete) wrote about [Binary Frameworks in Swift](https://pspdfkit.com/blog/2018/binary-frameworks-swift/)
->This article explores what ABI means and how it can be important for third-party frameworks.
-> <br/>TL;WR: ABI stability won’t change much for you, and it’s not enough to ship binary Swift frameworks.
+> This article explores what ABI means and how it can be important for third-party frameworks.
```
Comment (BasThomas) [inline _drafts/2018-02-08-issue-102.md]: Let's spell out `ABI` and put the full `Application Binary Interface` there.

File: _drafts/2018-02-08-issue-102.md [hunk 4]
```diff
@@ -29,24 +29,74 @@ author: tapthaker
...
+>
+> TL;WR: ABI stability won’t change much for you, and it’s not enough to ship binary Swift frameworks.
 
 ### Commits and pull requests
 
-[Doug Gregor](https://github.com/DougGregor) [fixed](https://github.com/apple/swift/pull/14174) an [issue](https://bugs.swift.org/browse/SR-6841) in type checker where class constraints `(spelled T: AnyObject)` on generic types were not getting checked on generic arguments.
+[Doug Gregor](https://github.com/DougGregor) has merged several pull requests:
+
+- [merged](https://github.com/apple/swift/pull/14296) a pull request that mangles retroactive conformances as part of bound generic types.
```
Comment (BasThomas) [inline _drafts/2018-02-08-issue-102.md]: Let's removed the `merged` here and put the link on `a pull request`.

File: _drafts/2018-02-08-issue-102.md [hunk 5]
```diff
@@ -29,24 +29,74 @@ author: tapthaker
...
+
+> A "retroactive" protocol conformance is a conformance that is provided
```
Comment (BasThomas) [inline _drafts/2018-02-08-issue-102.md]: I think we should shorten (or even completely remove) these excerpts. That would also help trim down the issue. :)

File: _drafts/2018-02-08-issue-102.md [hunk 6]
```diff
@@ -29,24 +29,74 @@ author: tapthaker
...
+the generic type. This prevents name collisions between (e.g.) types
+formed using retroactive conformances from different modules. The
+impact on the size of the mangling is expected to be relatively small,
+because most conformances are not retroactive.
+
+- [merged]((https://github.com/apple/swift/pull/14327)) a pull request that dynamically checks generic constraints when forming type info (e.g., Any.Type) from a mangled name.
+
+> Extend support for mapping a mangled name to type metadata to include support for checking protocol conformance requirements, using the encoding of generic requirements that is now available within context descriptors. For example, this allows `_typeByMangledName(mangled-name-of-Set<Int>)` to construct proper type metadata, filling in the `Int: Hashable` requirement as appropriate.
+
+- [merged](https://github.com/apple/swift/pull/14368) a pull request that will allow querying of conditional conformances at runtime. This was the last major part of implementing [SE-0143](https://github.com/apple/swift-evolution/blob/master/proposals/0143-conditional-conformances.md) *Conditional conformances* which is now complete 🎉.
```
Comment (BasThomas) [inline _drafts/2018-02-08-issue-102.md]: SE-0143 is also a significant part of reaching ABI; can we add something like: `getting us a step closer towards ABI stability in the process`?

File: _drafts/2018-02-08-issue-102.md [hunk 7]
```diff
@@ -29,24 +29,74 @@ author: tapthaker
...
+
+> Runtime query of conditional conformances is now implemented. Therefore, a dynamic cast such as value as? P, where the dynamic type of value conditional conforms to P, will succeed when the conditional requirements are met.
+
+- He also [fixed](https://github.com/apple/swift/pull/14174) an [issue](https://bugs.swift.org/browse/SR-6841) in type checker where class constraints `(spelled T: AnyObject)` on generic types were not getting checked on generic arguments.
```
Comment (BasThomas) [inline _drafts/2018-02-08-issue-102.md]: `in the* type checker`

Comment (BasThomas) [reply _drafts/2018-02-08-issue-102.md]: I guess we can also add this to the whole list of Doug's pull requests?

File: _drafts/2018-02-08-issue-102.md [hunk 8]
```diff
@@ -29,24 +29,74 @@ author: tapthaker
...
 
 [Greg Titus](https://github.com/gregomni) [merged](https://github.com/apple/swift/pull/14227) a pull request that removes compiler terminology l-value. As mentioned by Slava Pestov in this [tweet](https://twitter.com/slava_pestov/status/957720067822706688)
 > “Removing terminology” is almost always good. Compilers often use terminology that is internal to the implementation, not part of the language model, and can change.
 
-[Connor Wakamo](https://github.com/cwakamo) [proposed](https://github.com/apple/swift-evolution/pull/784) to 
-Replace PlaygroundQuickLook/CustomPlaygroundQuickLookable with new API
-> This proposal has been discussed on the mailing list/forum, and is now ready for official review. As noted in the proposal, parts of it are targeted at Swift 4.1 (deprecation of the old API and introduction of the new API) and others are targeted at Swift 5 (removal of the old API and introduction of a playground-specific shim library).
+[Nathan Hawes](https://github.com/nathawes) Created a [pull request](https://github.com/apple/swift/pull/14353) to fix a crash in onDocumentUpdateNotification. This was one of the top SourceKit crashes.
```
Comment (BasThomas) [inline _drafts/2018-02-08-issue-102.md]: `Created` can be lowercased (`created`).

Comment (BasThomas) [reply _drafts/2018-02-08-issue-102.md]: Let's add backticks around `onDocumentUpdateNotification`, to clarify that it is code.

File: _drafts/2018-02-08-issue-102.md [hunk 9]
```diff
@@ -29,24 +29,74 @@ author: tapthaker
...
+
+[Morten Bek Ditlevsen](https://github.com/mortenbekditlevsen) added Conditional conformances to `Hashable` for:
```
Comment (BasThomas) [inline _drafts/2018-02-08-issue-102.md]: `for the following types:`

File: _drafts/2018-02-08-issue-102.md [hunk 10]
```diff
@@ -29,24 +29,74 @@ author: tapthaker
...
+- `CollectionOfOne`, `EmptyCollection` and `Range` types. [commit](https://github.com/apple/swift/commit/618df4aeac766fcb8069e90a44b867969a1bc47d)
```
Comment (BasThomas) [inline _drafts/2018-02-08-issue-102.md]: ... and then omit the `types` at the end of the sentence here. Reads a bit more intuitive.

Comment (BasThomas) [reply _drafts/2018-02-08-issue-102.md]: Nit @jessesquires: I am not sure, but do you normally use an Oxford comma or not?

File: _drafts/2018-02-08-issue-102.md [hunk 11]
```diff
@@ -29,24 +29,74 @@ author: tapthaker
...
+- `Optional`, `Dictionary` and `Array` types. [pull request](https://github.com/apple/swift/pull/14247)
+
+[Eric Eckstein](https://github.com/eeckstein) [fixed](https://github.com/apple/swift/pull/14338) two issues which prevents incremental LLVM compilation.
```
Comment (BasThomas) [inline _drafts/2018-02-08-issue-102.md]: `which prevented` as issues is plural.

Comment (BasThomas) [reply _drafts/2018-02-08-issue-102.md]: Let's also add a blank line between this and the quote.

Comment (BasThomas) [reply _drafts/2018-02-08-issue-102.md]: ... or remove the quote; I don't feel like it adds that much here. :)

File: _drafts/2018-02-08-issue-102.md [hunk 12]
```diff
@@ -29,24 +29,74 @@ author: tapthaker
...
+> - debug-info: Don't write temporary file names in the debug info.
+>
+> This would prevent incremental llvm compilation because we would generate different IR on every compiler invocation.
+>
+> - SILCombiner: fix a non-determinism in a peephole optimization for load instructions.
 
 ### Accepted proposals
 
-> TODO
+[SE-0196](https://github.com/apple/swift-evolution/blob/master/proposals/0196-diagnostic-directives.md): *Compiler Diagnostic Directives* has been accepted but with a minor revision.
```
Comment (BasThomas) [inline _drafts/2018-02-08-issue-102.md]: Let's make this `was accepted` to keep it in line with previous issues, as well as add the URL to the forum post (where the proposal was accepted) to it.

File: _drafts/2018-02-08-issue-102.md [hunk 13]
```diff
@@ -29,24 +29,74 @@ author: tapthaker
...
+
+> The only revision over the original proposal is to change the syntax to use `#warning(<Message>)` instead of `#warning <Messsage>`. This fits well with most of Swift’s existing compiler directives, and was strongly supported in the review discussion.
+>
+> The review discussion also covered a variety of possible extensions or variants to this proposal, including support for using #warning as an expression instead of a line directive and support for runtime issues. The Core Team decided that while these directions are interesting and worth exploring, they are complementary to the core functionality serviced by this proposal. Further, keeping #warning as a line directive allows it to be used in a wide variety of contexts, and serves a different need than using it as a placeholder expression.
```
Comment (BasThomas) [inline _drafts/2018-02-08-issue-102.md]: Backticks arounds `#warning` in its two occurrences here maybe? :)

File: _drafts/2018-02-08-issue-102.md [hunk 14]
```diff
@@ -29,24 +29,74 @@ author: tapthaker
...
+
+The proposal was also implemented and [merged](https://github.com/apple/swift-evolution/blob/master/proposals/0196-diagnostic-directives.md) to master. Kudos to [Harlan Haskins](https://github.com/harlanhaskins).
```
Comment (BasThomas) [inline _drafts/2018-02-08-issue-102.md]: We can omit `to master` here, I think.

File: _drafts/2018-02-08-issue-102.md [hunk 15]
```diff
@@ -29,24 +29,74 @@ author: tapthaker
...
+
+[SE-0197](https://forums.swift.org/t/se-0197-add-in-place-remove-where/8872): *Adding in-place `remove(where:)` to the Standard Library* was accepted.
```
Comment (BasThomas) [inline _drafts/2018-02-08-issue-102.md]: Let's again add the URL to the blog post where the proposal was accepted.

File: _drafts/2018-02-08-issue-102.md [hunk 16]
```diff
@@ -29,24 +29,74 @@ author: tapthaker
...
+
+> The review of this proposal is now over. The core team has decided to accept the proposal with the revision that the method be named `removeAll(where:)`.
+In this thread, we also discussed whether to add a `removeAll(_: Element)` method. While I think that’s a reasonable-sounding idea, I do think it can considered independently of SE-0197, since it wasn’t originally part of the pitch. If you’re interested in adding that method, I encourage you to start a new pitch thread for it.
```
Comment (BasThomas) [inline _drafts/2018-02-08-issue-102.md]: Let's link SE-0197 here :)

File: _drafts/2018-02-08-issue-102.md [hunk 17]
```diff
@@ -58,13 +108,73 @@ Replace PlaygroundQuickLook/CustomPlaygroundQuickLookable with new API
 
 ### Proposals in review
 
+[SE-0195](https://github.com/apple/swift-evolution/blob/master/proposals/0195-dynamic-member-lookup.md): *Introduce User-defined “Dynamic Member Lookup” Types*is still under review. It's review was extended by another week.
```
Comment (BasThomas) [inline _drafts/2018-02-08-issue-102.md]: We need a space after the `Types*` here.

File: _drafts/2018-02-08-issue-102.md [hunk 18]
```diff
@@ -58,13 +108,73 @@ Replace PlaygroundQuickLook/CustomPlaygroundQuickLookable with new API
...
+
+> The proposal is accepted in principle, but specific details of the proposal need to be further discussed and ironed out. Specifically, there is the matter of using a marker protocol, which raises a bunch of technical questions.
+>
+> On the general principle of the proposal, the Core Team felt that:
+>
+> This proposal added valuable functionality to Swift
+
```
Comment (BasThomas) [inline _drafts/2018-02-08-issue-102.md]: This + 119 + 120 are missing `>`s.

File: _drafts/2018-02-08-issue-102.md [hunk 19]
```diff
@@ -58,13 +108,73 @@ Replace PlaygroundQuickLook/CustomPlaygroundQuickLookable with new API
...
+This proposal is not at odds at potentially adding any new dynamic affordances to Swift later that (say) tie into Swift’s runtime metadata, etc.
+There are tooling affordances, such as syntax coloring, that can be used to distinguish methods call going through this member lookup mechanism — without adding additional syntactic weight that would be at odds of some of the core goals of this proposal.
+
+[SE-0198](https://github.com/apple/swift-evolution/blob/master/proposals/0198-playground-quicklook-api-revamp.md): *Playground QuickLook API Revamp is in review.*
```
Comment (BasThomas) [inline _drafts/2018-02-08-issue-102.md]: The `*` should end after `Revamp` - and let's add the URL to the blog post as well.

File: _drafts/2018-02-08-issue-102.md [hunk 20]
```diff
@@ -58,13 +108,73 @@ Replace PlaygroundQuickLook/CustomPlaygroundQuickLookable with new API
...
 [SE-0197](https://forums.swift.org/t/se-0197-add-in-place-remove-where/8872): Adding in-place `remove(where:)` to the Standard Library is in review.
+
 > It is common to want to remove all occurrences of a certain element from a collection. This proposal is to add a remove algorithm to the standard library, which will remove all entries in a collection in-place matching a given predicate.
 
 ### Swift Forums
 
-> TODO
+Discussions around Strings:
+
+- [Brent Royal-Gordon](https://github.com/brentdax) started a discussion on [String interpolation revamp](https://forums.swift.org/t/string-interpolation-revamp/9302) and is looking for some ideas to generate fewer temporary strings. Currenly in Swift interpolation, when you write code like this in Swift 4:
```
Comment (BasThomas) [inline _drafts/2018-02-08-issue-102.md]: `on a* String interpolation revamp`

File: _drafts/2018-02-08-issue-102.md [hunk 21]
```diff
@@ -58,13 +108,73 @@ Replace PlaygroundQuickLook/CustomPlaygroundQuickLookable with new API
...
+
+{% highlight swift %}
+let foo: MyString = "foo \(bar) baz"
+{% endhighlight %}
+
+We currently generate code like below which means we create reference-counted heap objects which are purely temporary and are discarded as soon as the string interpolation is finished.
```
Comment (BasThomas) [inline _drafts/2018-02-08-issue-102.md]: Is this a quote? In that case, can we prefix it with `> `?

File: _drafts/2018-02-08-issue-102.md [hunk 22]
```diff
@@ -58,13 +108,73 @@ Replace PlaygroundQuickLook/CustomPlaygroundQuickLookable with new API
...
+)
+{% endhighlight %}
+
+- [Matthew Johnson](https://github.com/anandabits) started a discussion on [Generalized pattern matching](https://forums.swift.org/t/generalized-pattern-matching/9191).
+
+[Ben Cohen](https://github.com/airspeedswift) started a discussion for [Revisiting the choice of sort algorithm](https://forums.swift.org/t/revisiting-the-choice-of-sort-algorithm/8958)
+
+> `Swift.sort` is currently an [introsort](https://en.wikipedia.org/wiki/Introsort). Introsort is unstable, and there’s been [discussion](https://forums.swift.org/t/add-stable-sort-algorithm/778) over on evolution of adding a stable sort, either instead of or as well as the current sort.
+>
+> The current sort is problematic because it uses recursion heavily, which defeats a number of optimizations relating to ARC and elimination of the overhead of passing in a closure for the comparative for the comparator. This is why we use gyb to stamp out two near identical versions rather than implementing the Equatable version in terms of the closure-taking one.
```
Comment (BasThomas) [inline _drafts/2018-02-08-issue-102.md]: Lets backtick `gyb` as well as `Equatable`.

File: _drafts/2018-02-08-issue-102.md [hunk 23]
```diff
@@ -58,13 +108,73 @@ Replace PlaygroundQuickLook/CustomPlaygroundQuickLookable with new API
...
+
+[Jordan Rose](https://github.com/jrose-apple) started a discussion for [`@_exported` and fixing import visibility](https://forums.swift.org/t/exported-and-fixing-import-visibility/9415).
+> Today, if your Swift library “Foo” uses a library “Bar”, the headers (or swiftmodule files) of “Bar” must be available to all clients of “Foo”. This is clearly undesirable, especially when “Bar” is just an implementation detail. (It may even be linked statically.)
```
Comment (BasThomas) [inline _drafts/2018-02-08-issue-102.md]: There should be an empty line before the quote here.

File: _drafts/2018-02-08-issue-102.md [hunk 24]
```diff
@@ -58,13 +108,73 @@ Replace PlaygroundQuickLook/CustomPlaygroundQuickLookable with new API
...
 
 ### Finally
 
-> TODO: something funny/fun. tweet, link, etc.
+Let's look at the [change log](https://twitter.com/jckarter/status/959091909129113602) for Swift 5 😃🤣
```
Comment (BasThomas) [inline _drafts/2018-02-08-issue-102.md]: `change log` can be one word (`changelog`) here.

**Comment by BasThomas** [comment]
Two more small things from the previous pull request:

>  [Swift Bi-Weekly Brief](https://spec.fm/podcasts/swift-unwrapped/108765)

Can we add a description of the podcast's episode there?

> - [SR-6852](https://bugs.swift.org/browse/SR-6852) Support `#if swift(<4.1)` *#compiler*

I don't think we've ever added these hashes before - what was the reason for adding them?

File: _drafts/2018-02-08-issue-102.md (hunk 12 above)
Comment (tapthaker) [reply _drafts/2018-02-08-issue-102.md]: @BasThomas Do you mean like [was accepted] to Swift forums ?

**Comment by tapthaker** [comment]
>Can we add a description of the podcast's episode there?
Written in the latest draft update
>I don't think we've ever added these hashes before - what was the reason for adding them?
That was a trial I wanted to talk about but forgot. Basically the idea is that the readers can understand what section does this starter bug belong to.

File: _drafts/2018-02-08-issue-102.md (hunk 12 above)
Comment (BasThomas) [reply _drafts/2018-02-08-issue-102.md]: Yep!

File: _drafts/2018-02-08-issue-102.md (hunk 21 above)
Comment (tapthaker) [reply _drafts/2018-02-08-issue-102.md]: @BasThomas  Its not exactly the quote.

**Comment by tapthaker** [comment]
@BasThomas  @jessesquires : We need lint rules for 
- Full sentences
- Quotes

**Comment by tapthaker** [comment]
@BasThomas : I have no idea for intro comments, I need your help with that.

**Comment by jessesquires** [comment]
For the starter bugs, we do this:

> - SR-6852 [Compiler] Add support for `#if swift(<4.1)`

Also, we should adjust the bug title if needed. In this case, we should expand to a more complete sentence.

**Comment by jessesquires** [comment]
@tapthaker For intro, I suggest looking back through previous issues to get an idea of what usually goes here. 😄

**Comment by tapthaker** [comment]
@BasThomas @jessesquires : will be making the last few changes in the next 3-4 hours. Let me know if you want me to modify anything.

**Comment by BasThomas** [comment]
Looks good, will do another quick review later today and then I'll publish :)

**Comment by jessesquires** [comment]
@tapthaker the proposal numbers should link to the proposal itself, not the swift forums 😊

**Comment by jessesquires** [comment]
i'll push some fixes here, then merge

**Comment by jessesquires** [comment]
@tapthaker - I dont see any intro comments... ? Did you forget to push?

**Pull request merged**